src/rdkafka_admin.c

```diff
@@ -9009,3 +9010,256 @@ void rd_kafka_DescribeCluster(rd_kafka_t *rk,
+                                    rd_kafka_buf_t *reply,
+                                    char *errstr,
+                                    size_t errstr_size) {
+        const int log_decode_errors           = LOG_ERR;
+        rd_kafka_op_t *rko_result             = NULL;
+        rd_kafka_ElectLeadersResult_t *result = NULL;
+        int16_t top_level_error_code          = 0;
+        int32_t TopicArrayCnt;
+        rd_list_t partitions_arr;
+        int i;
+        int idx;
+
+        rd_kafka_buf_read_throttle_time(reply);
+
+        if (rd_kafka_buf_ApiVersion(reply) >= 1) {
+                rd_kafka_buf_read_i16(reply, &top_level_error_code);
+        }
+
+        /* #partitions */
+        rd_kafka_buf_read_arraycnt(reply, &TopicArrayCnt, RD_KAFKAP_TOPICS_MAX);
+
+        rd_list_init(&partitions_arr, TopicArrayCnt,
+                     rd_kafka_topic_partition_result_free);
+
+        for (i = 0; i < TopicArrayCnt; i++) {
+                rd_kafka_topic_partition_result_t *partition_result;
+                rd_kafkap_str_t ktopic;
+                char *topic;
+                int32_t partition;
+                int16_t partition_error_code;
+                rd_kafkap_str_t partition_error_msg;
+                char *partition_errstr;
+                int32_t PartArrayCnt;
+
+                rd_kafka_buf_read_str(reply, &ktopic);
+                RD_KAFKAP_STR_DUPA(&topic, &ktopic);
+
+                rd_kafka_buf_read_arraycnt(reply, &PartArrayCnt,
+                                           RD_KAFKAP_PARTITIONS_MAX);
+
+                for (idx = 0; idx < PartArrayCnt; idx++) {
+                        rd_kafka_buf_read_i32(reply, &partition);
+                        rd_kafka_buf_read_i16(reply, &partition_error_code);
+                        rd_kafka_buf_read_str(reply, &partition_error_msg);
+
+                        rd_kafka_buf_skip_tags(reply);
+
+                        if (partition_error_code) {
+                                if (RD_KAFKAP_STR_IS_NULL(
+                                        &partition_error_msg) ||
+                                    RD_KAFKAP_STR_LEN(&partition_error_msg) ==
+                                        0)
+                                        partition_errstr =
+                                            (char *)rd_kafka_err2str(
+                                                partition_error_code);
+                                else
+                                        RD_KAFKAP_STR_DUPA(
+                                            &partition_errstr,
+                                            &partition_error_msg);
+                        }
+                        partition_result = rd_kafka_topic_partition_result_new(
+                            topic, partition, partition_error_code,
+                            partition_errstr);
+
+                        rd_list_add(&partitions_arr, partition_result);
```

Comment:
We need to put the partition result in the original position for the request. Please check the implementation for the same in other APIs and do the same here.